File: routes/admin/api/status.go
package api

import (
	"encoding/json"
	"github.com/bamzi/jobrunner"
	"jinya-fonts/config"
	"jinya-fonts/storage"
	"net/http"
	"time"
)

type status struct {
	JobIsScheduled   bool       `json:"jobIsScheduled"`
	JobNextExecution *time.Time `json:"jobNextExecution"`
	Online           bool       `json:"online"`
	RedisUp          bool       `json:"redisUp"`
	ServingWebsite   bool       `json:"servingWebsite"`
}

func getStatus(w http.ResponseWriter, _ *http.Request) {
	currentStatus := status{
		JobIsScheduled: len(jobrunner.Entries()) > 0,
		Online:         true,
		RedisUp:        storage.CheckRedis(),
		ServingWebsite: config.LoadedConfiguration.ServeWebsite,
	}

	if currentStatus.JobIsScheduled {
		jobrunnerStatus := jobrunner.StatusPage()
		if len(jobrunnerStatus) > 0 {
			currentStatus.JobNextExecution = &jobrunnerStatus[0].Next
		}
	}

	err := json.NewEncoder(w).Encode(currentStatus)
	if err != nil {
		http.Error(w, "Failed to encode body", http.StatusInternalServerError)
	}
}

func getHealth(w http.ResponseWriter, r *http.Request) {
	_, _ = w.Write([]byte("OK"))
}

File: static/admin/js/settings/status.js
import { Alpine } from '../../../lib/alpine.js';
import { get } from '../../../lib/jinya-http.js';

import '../../lib/ui/online-indicator.js';

Alpine.data('statusData', () => ({
  jobIsScheduled: false,
  jobNextExecution: '',
  online: false,
  redisUp: false,
  mongoUp: false,
  servingWebsite: false,
  loading: true,
  async init() {
    const status = await get('/api/admin/status');
    this.jobIsScheduled = status.jobIsScheduled;
    this.jobNextExecution = new Date(Date.parse(status.jobNextExecution));
    this.online = status.online;
    this.redisUp = status.redisUp;
    this.servingWebsite = status.servingWebsite;
    this.loading = false;
  },
}));
